components/nimbus/src/enrollment.rs

```diff
@@ -373,40 +375,72 @@ impl ExperimentEnrollment {
-            ),
+            } => {
+                nimbus_events::enrollment.record(
+                    nimbus_events::EnrollmentExtra {
+                        branch: Some(branch.clone()),
+                        enrollment_id: Some(enrollment_id.clone().to_string()),
+                        experiment: Some(self.slug.clone())
+                    }
+                );
+                EnrollmentChangeEvent::new(
+                    &self.slug,
+                    enrollment_id,
+                    branch,
+                    None,
+                    EnrollmentChangeEventType::Enrollment,
+                )
+            },
             EnrollmentStatus::WasEnrolled {
                 enrollment_id,
                 branch,
                 ..
-            } => EnrollmentChangeEvent::new(
-                &self.slug,
-                enrollment_id,
-                branch,
-                None,
-                EnrollmentChangeEventType::Unenrollment,
-            ),
+            } => {
+                nimbus_events::unenrollment.record(
+                    nimbus_events::UnenrollmentExtra {
+                        branch: Some(branch.clone()),
+                        enrollment_id: Some(enrollment_id.clone().to_string()),
+                        experiment: Some(self.slug.clone())
+                    }
+                );
+                EnrollmentChangeEvent::new(
+                    &self.slug,
+                    enrollment_id,
+                    branch,
+                    None,
+                    EnrollmentChangeEventType::Unenrollment,
+                )
+            },
             EnrollmentStatus::Disqualified {
                 enrollment_id,
                 branch,
                 reason,
                 ..
-            } => EnrollmentChangeEvent::new(
-                &self.slug,
-                enrollment_id,
-                branch,
-                match reason {
-                    DisqualifiedReason::NotTargeted => Some("targeting"),
-                    DisqualifiedReason::OptOut => Some("optout"),
-                    DisqualifiedReason::Error => Some("error"),
-                },
-                EnrollmentChangeEventType::Disqualification,
-            ),
+            } => {
+                nimbus_events::disqualification.record(
+                    nimbus_events::DisqualificationExtra {
+                        branch: Some(branch.clone()),
+                        enrollment_id: Some(enrollment_id.clone().to_string()),
+                        experiment: Some(self.slug.clone()),
+                        reason: match reason {
+                            DisqualifiedReason::NotTargeted => Some("targeting".to_string()),
+                            DisqualifiedReason::OptOut => Some("optout".to_string()),
+                            DisqualifiedReason::Error => Some("error".to_string()),
+                        },
+                    }
+                );
```

Comment:
I'm confused about the differences in build and dependencies for iOS and Android (and Desktop). Is the intention here allow Glean to be conditionally compiled and then falling back to the recording events via existing foreign language bindings?

Reply:
Sorry, this has bit-rotted a bit and doesn't reflect conversations that we have had regarding how to go about letting iOS record in Rust while Android still uses the existing Glean Kotlin interface.

This was originally done to allow me to test the Glean changes I had made in a downstream consumer. I don't think we will need to worry about conditionally compiling Glean depending on platform, Glean "should just work" regardless of what language binding is being used, even if they are mix and matched in the same product (Nimbus recording in Rust while Fenix is recording in Kotlin or iOS in Swift, etc).

Reply:
What I don't know about is how we know we are working with the same version of Glean coming from GeckoView...